modules/apps/organizations/organizations-item-selector-web/src/main/java/com/liferay/organizations/item/selector/web/internal/display/context/OrganizationItemSelectorViewDisplayContext.java

```diff
@@ -75,16 +83,13 @@ public SearchContainer<Organization> getSearchContainer()
 			(OrganizationSearchTerms)_searchContainer.getSearchTerms();
 
 		_searchContainer.setResultsAndTotal(
-			() -> _organizationLocalService.search(
-				CompanyThreadLocal.getCompanyId(),
-				OrganizationConstants.ANY_PARENT_ORGANIZATION_ID,
-				organizationSearchTerms.getKeywords(), null, null, null, null,
-				_searchContainer.getStart(), _searchContainer.getEnd(),
-				_searchContainer.getOrderByComparator()),
-			_organizationLocalService.searchCount(
-				CompanyThreadLocal.getCompanyId(),
-				OrganizationConstants.ANY_PARENT_ORGANIZATION_ID,
-				organizationSearchTerms.getKeywords(), null, null, null, null));
+			_filterOrganizations(
+				_organizationLocalService.search(
+					CompanyThreadLocal.getCompanyId(),
+					OrganizationConstants.ANY_PARENT_ORGANIZATION_ID,
+					organizationSearchTerms.getKeywords(), null, null, null,
+					null, QueryUtil.ALL_POS, QueryUtil.ALL_POS,
+					_searchContainer.getOrderByComparator())));
```

Comment:
@victorcdp sorry but I don't like this because if you have 1000 organizations we are getting all of them to filter out the one I cannot view...

Try check how are retrieved by the API `OrganizationResourceImpl _getOrganizationsPage` and/or `ViewOrganizationsManagementToolbarDisplayContext getSearchContainer` ... I _think_ here are already filtered by the visibility